lib/galaxy/tools/data/__init__.py

```diff
@@ -616,13 +616,12 @@ def get_filename_for_source(self, source, default=None):
         # No toolshed data table entry was found, try the first file found on disk
         else:
             for filename in self.filenames.keys():
-                if os.path.exists(filename):
-                    fmt_dict = dict(source_repo_info)
-                    fmt_dict['filename'] = filename
+                if source_repo_info is not None:
```

Comment:
Was that intentional to drop the `os.path.exists` check ?

Reply:
Yeah, later in the code it creates the file if it doesn't exist, so that check is redundant.

Reply:
Fair point. I also am unsure how to handle situations where the _path_ to the filename doesn't exist.

Reply:
I don't know that either, but the os.path.exists approach seemed reasonable (maybe also check if the file is writable?), I think the switching between first and last item though may make debugging more difficult. Maybe add an else branch to this if branch and add appropriate logging there too and return the filename from both branches ?

Reply:
Good idea on the else, I'll add that too